llmfoundry/models/hf/hf_base.py

```diff
@@ -80,7 +80,14 @@ def __init__(
         additional_train_metrics: Optional[list] = None,
         additional_eval_metrics: Optional[list] = None,
         should_save_peft_only: bool = True,
+        attn_implementation: str = 'eager',
```

Comment:
this doesn't seem like quite the right behavior. I think attn_implementation should default to None, and then you warn if the user set both attn_implementation and use_flash_attention_2. Cause as is, you would immediately start getting warnings about attn_implementation being set incorrectly.